spec/requests/api/provision_requests_spec.rb

```diff
@@ -95,6 +95,149 @@
     end
   end
 
+  context "AWS advanced provision requests" do
+    let(:aws_dialog) do
+      dialog = FactoryGirl.create(:miq_dialog_aws_provision)
+      allow_any_instance_of(MiqRequestWorkflow).to receive(:dialog_name_from_automate).and_return(dialog.name)
+    end
+    let(:ems) { FactoryGirl.create(:ems_amazon_with_authentication) }
+    let(:template) do
+      FactoryGirl.create(:template_amazon, :name => "template1", :ext_management_system => ems)
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon, :ems_id => ems.id) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
+    let(:cloud_subnet1) do
+      FactoryGirl.create(:cloud_subnet, :ems_id => ems.id, :cloud_network => cloud_network1, :availability_zone => az)
+    end
+    let(:security_group1) do
+      FactoryGirl.create(:security_group_amazon, :name => "sgn_1", :ext_management_system => ems,
+                         :cloud_network => cloud_network1)
+    end
+    let(:floating_ip1) { FactoryGirl.create(:floating_ip, :cloud_network_only => true, :ext_management_system => ems) }
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {
+          "owner_first_name" => "John",
+          "owner_last_name"  => "Doe",
+          "owner_email"      => "user@example.com"
+        }
+      }
+    end
+
+    let(:provreq1_body) do
```

Comment:
if the new method is only merging an hash to an other, I think we can omit this method entirely and move those simply in the concerned test.
will try this way and you tell me if it makes sense to you :)